lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -490,7 +490,8 @@ export class UserAgentApplication {
                 scopes,
                 responseType,
                 this.getRedirectUri(),
-                request && request.state
+                request && request.state,
```

Comment:
We are passing three different parameters from the request object into ServerRequestParameters - scopes, state and now correlationId. May be better to pass the request to ServerRequestParameters as a whole object.

Reply:
Yes. Agreed.

Reply:
Added a new type. Please review.

Reply:
Changed it. Since `getTokenCachedInternal` is not being changed in Angular version for the immediate future, cannot pass request to the `ServerRequestParameters`.